### PR TITLE
`Action#format` should return only a value from `Action.accept`

### LINKED ISSUE #205: Getting `Can't find template 'home/index' for 'xml' format.` despite `accept :html` statement
Relevant headers:
```
Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,*/*;q=0.5
Accept-Charset: utf-8,*;q=0.7
Accept-Encoding: gzip
```

Exception:
```
Hanami::View::MissingTemplateError: Can't find template 'home/index' for 'xml' format.
  from hanami/view/rendering.rb:163:in `template'
  from hanami/view/rendering.rb:137:in `rendered'
  from hanami/view/rendering.rb:151:in `layout'
  from hanami/view/rendering.rb:105:in `render'
  from hanami/view/rendering.rb:256:in `render'
  from hanami/rendering_policy.rb:45:in `_render_action'
  from hanami/rendering_policy.rb:38:in `_render'
  from hanami/rendering_policy.rb:29:in `render'
  from hanami/application.rb:180:in `call'
  from delegate.rb:83:in `method_missing'
  from http_router.rb:193:in `process_destination_path'
  from (eval):15:in `call'
  from http_router.rb:288:in `raw_call'
  from hanami/routing/http_router.rb:155:in `raw_call'
  from hanami/router.rb:1007:in `call'
  from rack/methodoverride.rb:22:in `call'
  from rack/static.rb:124:in `call'
  from rack/builder.rb:153:in `call'
  from hanami/app.rb:41:in `call'
  from raven/integrations/rack.rb:50:in `call'
  from unicorn/http_server.rb:562:in `process_client'
  from unicorn/http_server.rb:658:in `worker_loop'
  from unicorn/http_server.rb:508:in `spawn_missing_workers'
  from unicorn/http_server.rb:132:in `start'
  from /foo/bar/baz/hash/vendor/bundle/ruby/2.3.0/gems/unicorn-5.1.0/bin/unicorn:126:in `<top (required)>'
  from bin/unicorn:17:in `load'
  from bin/unicorn:17:in `<main>'
```

```ruby
# Gemfile.lock (excerpt)

    hanami (0.9.2)
      bundler (~> 1.13)
      concurrent-ruby (~> 1.0)
      hanami-assets (~> 0.4)
      hanami-controller (~> 0.8)
      hanami-helpers (~> 0.5)
      hanami-mailer (~> 0.4)
      hanami-router (~> 0.8, >= 0.8.1)
      hanami-utils (~> 0.9)
      hanami-validations (~> 0.7)
      hanami-view (~> 0.8)
      thor (~> 0.19)
    hanami-assets (0.4.0)
      hanami-helpers (~> 0.5)
      hanami-utils (~> 0.9)
      tilt (~> 2.0, >= 2.0.2)
    hanami-controller (0.8.1)
      hanami-utils (~> 0.9)
      rack (~> 1.6, >= 1.6.2)
    hanami-helpers (0.5.1)
      hanami-utils (~> 0.9)
    hanami-mailer (0.4.0)
      hanami-utils (~> 0.9)
      mail (~> 2.5)
      tilt (~> 2.0, >= 2.0.1)
    hanami-model (0.7.0)
      concurrent-ruby (~> 1.0)
      dry-types (~> 0.9)
      hanami-utils (~> 0.8)
      rom-repository (~> 0.3)
      rom-sql (~> 0.9)
    hanami-router (0.8.1)
      hanami-utils (~> 0.9, >= 0.9.1)
      http_router (~> 0.11)
      rack (~> 1.6)
    hanami-utils (0.9.2)
    hanami-validations (0.7.1)
      dry-validation (~> 0.10)
      hanami-utils (~> 0.9)
    hanami-view (0.8.0)
      hanami-utils (~> 0.9)
      tilt (~> 2.0, >= 2.0.1)
```

Elided source:
```ruby
# apps/web/config/routes.rb

root to: 'home#index'
```
```ruby
# apps/web/controllers/home/index.rb

module Web::Controllers::Home
  class Index
    include Web::Action

    accept :html
end
```

### PR DESCRIPTION
**Pull request by jodosha**
Closes #205 

---

/cc @TrevorBramble @hanami/issues @hanami/ecosystem for review. Thanks!

### CODE REVIEW

**Comment by cllns** [comment]
LGTM! 👍

File: lib/hanami/controller/configuration.rb [hunk 1]
```diff
@@ -435,6 +435,18 @@ def mime_types
                         end
       end
 
+      # Restrict the MIME types set only to the given set
+      #
+      # @param mime_types [Array] the set of MIME types
+      #
+      # @since x.x.x
```
Comment (TrevorBramble) [inline lib/hanami/controller/configuration.rb]: Intentionally left unspecified?

**Comment by TrevorBramble** [comment]
Sorry, I tried using this branch to test locally (tediously setting up all of the hanami gems as github refs) but some changes to hanami/model are incompatible and I don't see an upgrade guide for 1.x yet.

Code looks great though!

File: lib/hanami/controller/configuration.rb (hunk 1 above)
Comment (cllns) [inline lib/hanami/controller/configuration.rb]: yup! that way we can do a repo-wide search for 'x.x.x' when we release. it gives us flexibility to decide the version number later :)

**Pull request merged**